### PR TITLE
Create STL and Cleanup Code

### PR DESCRIPTION
**Pull request by aidan-mundy**
This PR creates the Overclocked STL, a standard library of tools we can carry from year to year. Code that is generic enough to be used on any robot should go here, particularly if it may be of ongoing utility to the team. In the future we may move this STL into its own git repo, but for now it can remain in this one. 

I also did some refactoring of the constants file to make some of the variables less verbose and provide a more manageable structure.

### CODE REVIEW

File: src/main/java/overclocked/stl/io/OverclockedController.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+   * Creates a new OverclockedController.
+   *
+   * @param port The port index on the Driver Station that the device is plugged into.
+   */
+  public OverclockedController(int port) {
+    super(port);
+  }
+
+  /**
+   * Creates a new {@link JoystickButton} from this controller.
```
Comment (chit-sudeep) [inline src/main/java/overclocked/stl/io/OverclockedController.java:22]: Maybe it should say “new JoystickButton ‘for’ this controller.”?

File: src/main/java/overclocked/stl/io/OverclockedController.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+
+package overclocked.stl.io;
+
+import edu.wpi.first.wpilibj.GenericHID;
+import edu.wpi.first.wpilibj2.command.button.JoystickButton;
+
+/** 
+ * An extension of the {@link GenericHID} class that adds some useful utilities.
```
Comment (chit-sudeep) [inline src/main/java/overclocked/stl/io/OverclockedController.java:8]: There’s new utility, so maybe it should be singular (at least for now)?

**Review by chit-sudeep — CHANGES_REQUESTED** [review]
Please see my comments.

File: src/main/java/overclocked/stl/io/OverclockedController.java (hunk 1 above)
Comment (aidan-mundy) [reply src/main/java/overclocked/stl/io/OverclockedController.java:22]: The "from" is intentional, as the controller itself is an input into the JoystickButton constructor

File: src/main/java/overclocked/stl/io/OverclockedController.java (hunk 2 above)
Comment (aidan-mundy) [reply src/main/java/overclocked/stl/io/OverclockedController.java:8]: The plural is fine, saves a change later and doesn't really cause any confusion. "Adds a useful utility" sounds exceedingly vague and "Adds a utility that makes creating JoystickButtons cleaner" _is_ exceedingly verbose.

**Review by chit-sudeep — APPROVED** [review]
Looks good

**Review by kchenturtles — APPROVED** [review]
Seems all set.

**Pull request merged**